(3.3.8rc) Oracle error on AuthToken table, reserved word 'mode'

**Describe the bug**

Oracle fails to create table field:

`Error executing DDL "alter table auth_token add mode varchar2(255 char)" via JDBC Statement`
`Caused by: oracle.jdbc.OracleDatabaseException: ORA-00904: : invalid identifier`

**My Rundeck detail**

* Rundeck version: 3.3.8rc5
* DB Type/version: Oracle

**To Reproduce**
Steps to reproduce the behavior:
1. Startup

**Expected behavior**

No error

--- a/rundeckapp/grails-app/domain/rundeck/AuthToken.groovy
+++ b/rundeckapp/grails-app/domain/rundeck/AuthToken.groovy
@@ -33,7 +33,7 @@ class AuthToken implements AuthenticationToken {
     Date lastUpdated
     String name
     AuthTokenType type = AuthTokenType.USER
-    AuthTokenMode mode = AuthTokenMode.SECURED
+    AuthTokenMode tokenMode = AuthTokenMode.SECURED
     private transient String clearToken = null
 
     static belongsTo = [user:User]
@@ -49,7 +49,7 @@ class AuthToken implements AuthenticationToken {
         dateCreated(nullable: true)
         type(nullable: true)
         name(nullable: true)
-        mode(nullable: true)
+        tokenMode(nullable: true)
     }
     static mapping = {
         authRoles type: 'text'
@@ -67,16 +67,16 @@ class AuthToken implements AuthenticationToken {
 
 
     /**
-     * Encodes the token value according to the mode set.
+     * Encodes the token value according to the tokenMode set.
      */
     private void encodeToken() {
         this.clearToken = token
-        this.mode = this.mode ?: AuthTokenMode.LEGACY
-        this.token = encodeTokenValue(this.clearToken, this.mode)
+        this.tokenMode = this.tokenMode ?: AuthTokenMode.LEGACY
+        this.token = encodeTokenValue(this.clearToken, this.tokenMode)
     }
 
     /**
-     * Encodes a clear token value acording to the mode supplied.
+     * Encodes a clear token value acording to the tokenMode supplied.
      */
     public static String encodeTokenValue(String clearValue, AuthTokenMode mode){
         if(!clearValue)
@@ -102,13 +102,13 @@ class AuthToken implements AuthenticationToken {
         return createCriteria().get {
             or {
                 and {
-                    eq("mode", AuthTokenMode.SECURED)
+                    eq("tokenMode", AuthTokenMode.SECURED)
                     eq("token", tokenHash)
                 }
                 and {
                     or {
-                        isNull("mode")
-                        eq("mode", AuthTokenMode.LEGACY)
+                        isNull("tokenMode")
+                        eq("tokenMode", AuthTokenMode.LEGACY)
                     }
                     eq("token", tokenValue)
                 }
@@ -129,13 +129,13 @@ class AuthToken implements AuthenticationToken {
             eq("type", tokenType)
             or {
                 and {
-                    eq("mode", AuthTokenMode.SECURED)
+                    eq("tokenMode", AuthTokenMode.SECURED)
                     eq("token", tokenHash)
                 }
                 and {
                     or {
-                        isNull("mode")
-                        eq("mode", AuthTokenMode.LEGACY)
+                        isNull("tokenMode")
+                        eq("tokenMode", AuthTokenMode.LEGACY)
                     }
                     eq("token", tokenValue)
                 }

--- a/rundeckapp/grails-app/services/rundeck/services/ApiService.groovy
+++ b/rundeckapp/grails-app/services/rundeck/services/ApiService.groovy
@@ -20,7 +20,6 @@ import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenMode
 import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenType
 import com.dtolabs.rundeck.core.authentication.tokens.AuthenticationToken
 import com.dtolabs.rundeck.core.authentication.tokens.SimpleTokenBuilder
-import com.dtolabs.rundeck.core.authorization.AuthContextProvider
 import com.dtolabs.rundeck.core.authorization.AuthorizationUtil
 import com.dtolabs.rundeck.core.authorization.UserAndRolesAuthContext
 import com.dtolabs.rundeck.core.authorization.Validation
@@ -121,15 +120,15 @@ class ApiService {
         }
 
         AuthToken token = new AuthToken(
-                token: newtoken,
-                authRoles: AuthToken.generateAuthRoles(roles),
-                user: ownerUser,
-                expiration: expiration,
-                uuid: uuid,
-                creator: tokenData.creator,
-                name: tokenData.name,
-                type: tokenType,
-                mode: tokenMode
+            token: newtoken,
+            authRoles: AuthToken.generateAuthRoles(roles),
+            user: ownerUser,
+            expiration: expiration,
+            uuid: uuid,
+            creator: tokenData.creator,
+            name: tokenData.name,
+            type: tokenType,
+            tokenMode: tokenMode
         )
 
         if (token.save(flush:true)) {

--- a/rundeckapp/src/main/groovy/com/dtolabs/rundeck/app/api/tokens/Token.groovy
+++ b/rundeckapp/src/main/groovy/com/dtolabs/rundeck/app/api/tokens/Token.groovy
@@ -73,8 +73,8 @@ class Token {
         this.name = authToken.name
         this.id = authToken.uuid ?: authToken.id
         this.token = masked ? null :
-                (authToken.mode == null || authToken.mode == AuthTokenMode.LEGACY) ? authToken.token :
-                        authToken.clearToken
+                     (authToken.tokenMode == null || authToken.tokenMode == AuthTokenMode.LEGACY) ? authToken.token :
+                     authToken.clearToken
         this.v18TokenId = this.token
         this.creator = authToken.creator
         this.user = authToken.user.login

--- a/rundeckapp/src/test/groovy/rundeck/AuthTokenSpec.groovy
+++ b/rundeckapp/src/test/groovy/rundeck/AuthTokenSpec.groovy
@@ -19,7 +19,6 @@ package rundeck
 import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenMode
 import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenType
 import grails.testing.gorm.DataTest
-import grails.testing.gorm.DomainUnitTest
 import org.apache.commons.codec.digest.DigestUtils
 import spock.lang.Specification
 
@@ -121,10 +120,10 @@ class AuthTokenSpec extends Specification implements DataTest {
         given:
         def user = new User(login: "admin").save()
         def newToken = new AuthToken(
-                token: clear,
-                user: user,
-                authRoles: "admin",
-                mode: mode
+            token: clear,
+            user: user,
+            authRoles: "admin",
+            tokenMode: mode
         );
         def resultOnMem = newToken.save(flush: true, failOnError: true)
 
@@ -149,11 +148,11 @@ class AuthTokenSpec extends Specification implements DataTest {
         given:
         def user = new User(login: "admin").save()
         def newToken = new AuthToken(
-                user: user,
-                authRoles: "admin",
-                token: TOKEN_CLEAR_VALUE,
-                type: AuthTokenType.USER,
-                mode: AuthTokenMode.SECURED
+            user: user,
+            authRoles: "admin",
+            token: TOKEN_CLEAR_VALUE,
+            type: AuthTokenType.USER,
+            tokenMode: AuthTokenMode.SECURED
         )
         newToken.save(flush: true, failOnError: true)
         def tokFromDb = AuthToken.findById(newToken.id)
@@ -182,32 +181,32 @@ class AuthTokenSpec extends Specification implements DataTest {
         def tokenUnsec2 = "d8up8y43p93ny2rp9dw8nyu3pt98mw4uyt89wn58"
 
         def newToken1 = new AuthToken(
-                user: user,
-                uuid: "sec1",
-                authRoles: "admin",
-                token: tokenSec1,
-                mode: AuthTokenMode.SECURED
+            user: user,
+            uuid: "sec1",
+            authRoles: "admin",
+            token: tokenSec1,
+            tokenMode: AuthTokenMode.SECURED
         )
         def newToken2 = new AuthToken(
-                user: user,
-                uuid: "sec2",
-                authRoles: "admin",
-                token: tokenSec2,
-                mode: AuthTokenMode.SECURED
+            user: user,
+            uuid: "sec2",
+            authRoles: "admin",
+            token: tokenSec2,
+            tokenMode: AuthTokenMode.SECURED
         )
         def newToken3 = new AuthToken(
-                user: user,
-                uuid: "unsec1",
-                authRoles: "admin",
-                token: tokenUnsec1,
-                mode: AuthTokenMode.LEGACY
+            user: user,
+            uuid: "unsec1",
+            authRoles: "admin",
+            token: tokenUnsec1,
+            tokenMode: AuthTokenMode.LEGACY
         )
         def newToken4 = new AuthToken(
-                user: user,
-                uuid: "unsec2",
-                authRoles: "admin",
-                token: tokenUnsec2,
-                mode: AuthTokenMode.LEGACY
+            user: user,
+            uuid: "unsec2",
+            authRoles: "admin",
+            token: tokenUnsec2,
+            tokenMode: AuthTokenMode.LEGACY
         )
         newToken1.save(flush: true, failOnError: true)
         newToken2.save(flush: true, failOnError: true)

--- a/rundeckapp/src/test/groovy/rundeck/controllers/ApiControllerSpec.groovy
+++ b/rundeckapp/src/test/groovy/rundeck/controllers/ApiControllerSpec.groovy
@@ -20,14 +20,12 @@ import com.dtolabs.rundeck.app.api.ApiMarshallerRegistrar
 import com.dtolabs.rundeck.app.api.ApiVersions
 import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenMode
 import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenType
-import com.dtolabs.rundeck.core.authorization.AuthContextProvider
 import com.dtolabs.rundeck.core.authorization.UserAndRolesAuthContext
 import grails.converters.JSON
 import grails.converters.XML
 import grails.testing.gorm.DataTest
 import grails.testing.web.controllers.ControllerUnitTest
 import grails.web.mapping.LinkGenerator
-import org.rundeck.app.authorization.AppAuthContextEvaluator
 import org.rundeck.app.authorization.AppAuthContextProcessor
 import rundeck.AuthToken
 import rundeck.User
@@ -62,14 +60,14 @@ class ApiControllerSpec extends Specification implements ControllerUnitTest<ApiC
                 )
         createdToken.save()
         AuthToken webhookToken = new AuthToken(
-                user: bob,
-                type: AuthTokenType.WEBHOOK,
-                mode: AuthTokenMode.LEGACY,
-                token: 'whk',
-                authRoles: 'a,b',
-                uuid: '123uuidwhk',
-                creator: 'elf',
-                )
+            user: bob,
+            type: AuthTokenType.WEBHOOK,
+            tokenMode: AuthTokenMode.LEGACY,
+            token: 'whk',
+            authRoles: 'a,b',
+            uuid: '123uuidwhk',
+            creator: 'elf',
+            )
         webhookToken.save()
 
         controller.apiService = Mock(ApiService) {

--- a/rundeckapp/src/test/groovy/rundeck/interceptors/SetUserInterceptorSpec.groovy
+++ b/rundeckapp/src/test/groovy/rundeck/interceptors/SetUserInterceptorSpec.groovy
@@ -6,7 +6,6 @@ import com.dtolabs.rundeck.core.authentication.Username
 import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenMode
 import com.dtolabs.rundeck.core.authentication.tokens.AuthTokenType
 import com.dtolabs.rundeck.core.authentication.tokens.AuthenticationToken
-import grails.core.GrailsApplication
 import grails.testing.gorm.DataTest
 import grails.testing.web.interceptor.InterceptorUnitTest
 import rundeck.AuthToken
@@ -19,9 +18,7 @@ import rundeck.services.ConfigurationService
 import rundeck.services.UserService
 import spock.lang.Specification
 
-import javax.security.auth.Subject
 import javax.servlet.ServletContext
-import java.security.Principal
 
 class SetUserInterceptorSpec extends Specification implements InterceptorUnitTest<SetUserInterceptor>, DataTest {
 
@@ -114,10 +111,10 @@ class SetUserInterceptorSpec extends Specification implements InterceptorUnitTes
         User u1 = new User(login: "admin")
         User u2 = new User(login: "whk")
         AuthToken userTk1 = new AuthToken(token: "123",user:u1,authRoles:"admin",type: null)
-        AuthToken userTk2 = new AuthToken(token: "456",user:u1,authRoles:"admin",type: AuthTokenType.USER, mode: AuthTokenMode.LEGACY)
-        AuthToken userTk3 = new AuthToken(token: "ABC",user:u1,authRoles:"admin",type: AuthTokenType.USER, mode: AuthTokenMode.SECURED)
-        AuthToken userTk4 = new AuthToken(token: "DEF",user:u1,authRoles:"admin",type: AuthTokenType.USER, mode: null)
-        AuthToken whkTk = new AuthToken(token: "789",user:u2,authRoles:"admin",type:AuthTokenType.WEBHOOK, mode: AuthTokenMode.LEGACY)
+        AuthToken userTk2 = new AuthToken(token: "456", user:u1, authRoles:"admin", type: AuthTokenType.USER, tokenMode: AuthTokenMode.LEGACY)
+        AuthToken userTk3 = new AuthToken(token: "ABC", user:u1, authRoles:"admin", type: AuthTokenType.USER, tokenMode: AuthTokenMode.SECURED)
+        AuthToken userTk4 = new AuthToken(token: "DEF", user:u1, authRoles:"admin", type: AuthTokenType.USER, tokenMode: null)
+        AuthToken whkTk = new AuthToken(token: "789", user:u2, authRoles:"admin", type:AuthTokenType.WEBHOOK, tokenMode: AuthTokenMode.LEGACY)
         u1.save()
         u2.save()
         userTk1.save()

--- a/rundeckapp/src/test/groovy/rundeck/services/RundeckAuthTokenManagerServiceSpec.groovy
+++ b/rundeckapp/src/test/groovy/rundeck/services/RundeckAuthTokenManagerServiceSpec.groovy
@@ -37,11 +37,11 @@ class RundeckAuthTokenManagerServiceSpec extends Specification
             User user1 = new User(login: 'auser')
             user1.save()
             AuthToken existing = new AuthToken(
-                    token: '123',
-                    authRoles: 'a,b',
-                    user: user1,
-                    type: AuthTokenType.WEBHOOK,
-                    mode: AuthTokenMode.LEGACY
+                token: '123',
+                authRoles: 'a,b',
+                user: user1,
+                type: AuthTokenType.WEBHOOK,
+                tokenMode: AuthTokenMode.LEGACY
             )
             existing.save(flush: true)
             def auth = Mock(UserAndRolesAuthContext)

